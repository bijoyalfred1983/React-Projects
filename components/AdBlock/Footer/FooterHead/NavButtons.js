/* @flow */

import React from 'react';
import { Link } from 'react-router-dom';
import type { Element } from 'react';
import styles from './styles.scss';
import { jsTrack } from '../../../containers/Analytics/ClickTrack';

type Props = { btnContainer: Array<Object> };

export const trackFooterLink = (e, linkName) => {
  jsTrack.trackFooterNavMenu(linkName);
};

const NavButtons = ({ btnContainer }: Props): Element<'div'> => (
  <div className={styles.footerTop}>
    <ul className={styles.liWidthPrint}>
      {btnContainer.map(navbutton => (
        <li className={`col m4 s12 ${navbutton.className}`} key={navbutton.id}>
          <Link
            to={navbutton.linkTo}
            onClick={e => trackFooterLink(e, navbutton.content)}
          >
            <img src={navbutton.iconUrl} alt="not found" />
            <span>{navbutton.content.toUpperCase()}</span>
            <img
              src="/images/footer/right.svg"
              alt=""
              className={`hide-on-small-only ${styles.rightArrow}`}
            />
          </Link>
        </li>
      ))}
    </ul>
  </div>
);

export default NavButtons;
