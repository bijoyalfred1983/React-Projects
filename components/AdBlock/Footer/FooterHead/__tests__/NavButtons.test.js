/* @flow */
import React from 'react';
import { mount } from 'enzyme';
import { Link, StaticRouter } from 'react-router-dom';
import toJson from 'enzyme-to-json';
import NavButtons from '../NavButtons';

describe('<NavButtons />', () => {
  let PROPS;
  beforeEach(() => {
    PROPS = {
      btnContainer:  [
        {
          id: '1',
          linkTo: '/',
          content: 'locate a store',
          iconUrl: '/images/footer/location-pin.svg',
          className: ''
        },
        {
          id: '2',
          linkTo: '/trackOrder',
          content: 'track your order',
          iconUrl: '/images/footer/box.svg',
          className: ''
        }
      ]
    };
  });

  test('renders on NavButtons correctly', () => {
    const wrapper = mount(
      <StaticRouter context={{}}>
        <NavButtons {...PROPS} />
        </StaticRouter>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});



