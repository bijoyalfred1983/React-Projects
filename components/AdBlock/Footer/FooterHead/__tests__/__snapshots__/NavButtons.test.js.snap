// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavButtons /> renders on NavButtons correctly 1`] = `
<StaticRouter
  basename=""
  context={Object {}}
  location="/"
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <NavButtons
      btnContainer={
        Array [
          Object {
            "className": "",
            "content": "locate a store",
            "iconUrl": "/images/footer/location-pin.svg",
            "id": "1",
            "linkTo": "/",
          },
          Object {
            "className": "",
            "content": "track your order",
            "iconUrl": "/images/footer/box.svg",
            "id": "2",
            "linkTo": "/trackOrder",
          },
        ]
      }
    >
      <div
        className="footerTop"
      >
        <ul
          className=""
        >
          <li
            className="col m4 s12 "
          >
            <Link
              replace={false}
              to="/"
            >
              <a
                href="/"
                onClick={[Function]}
              >
                <img
                  alt="not found"
                  src="/images/footer/location-pin.svg"
                />
                <span>
                  LOCATE A STORE
                </span>
                <img
                  alt=""
                  className="hide-on-small-only rightArrow"
                  src="/images/footer/right.svg"
                />
              </a>
            </Link>
          </li>
          <li
            className="col m4 s12 "
          >
            <Link
              replace={false}
              to="/trackOrder"
            >
              <a
                href="/trackOrder"
                onClick={[Function]}
              >
                <img
                  alt="not found"
                  src="/images/footer/box.svg"
                />
                <span>
                  TRACK YOUR ORDER
                </span>
                <img
                  alt=""
                  className="hide-on-small-only rightArrow"
                  src="/images/footer/right.svg"
                />
              </a>
            </Link>
          </li>
        </ul>
      </div>
    </NavButtons>
  </Router>
</StaticRouter>
`;
